### PR TITLE
SONARJS-678 Extend DoubleDispatchVisitor: visitToken and visitComment

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: javascript-frontend/src/test/java/org/sonar/plugins/javascript/api/visitors/DoubleDispatchVisitorTest.java [hunk 1]
```diff
@@ -25,44 +25,68 @@
...
+import org.sonar.plugins.javascript.api.tree.lexical.SyntaxToken;
+import org.sonar.plugins.javascript.api.tree.lexical.SyntaxTrivia;
 
 import static org.fest.assertions.Assertions.assertThat;
 
 public class DoubleDispatchVisitorTest {
 
   @Test
+  public void test_visit_token() throws Exception {
+    assertNumberOfVisitedTokens("[1, 2, , ,]", 9);
```
Comment (pynicolas) [inline javascript-frontend/src/test/java/org/sonar/plugins/javascript/api/visitors/DoubleDispatchVisitorTest.java]: @vilchik-elena I see only 8 tokens: what's the trick?

File: javascript-frontend/src/main/java/org/sonar/plugins/javascript/api/visitors/DoubleDispatchVisitor.java [hunk 2]
```diff
@@ -125,439 +128,399 @@ protected void scan(@Nullable Tree tree) {
     }
   }
 
-  protected <T> void scan(List<T> trees) {
-    for (T tree : trees) {
+  protected void scanTree(Tree tree) {
```
Comment (pynicolas) [inline javascript-frontend/src/main/java/org/sonar/plugins/javascript/api/visitors/DoubleDispatchVisitor.java]: @vilchik-elena Would `scanChildren` be a better name?

File: javascript-frontend/src/test/java/org/sonar/plugins/javascript/api/visitors/DoubleDispatchVisitorTest.java (hunk 1 above)
Comment (vilchik-elena) [inline javascript-frontend/src/test/java/org/sonar/plugins/javascript/api/visitors/DoubleDispatchVisitorTest.java]: @pynicolas yes :)

File: javascript-frontend/src/main/java/org/sonar/plugins/javascript/api/visitors/DoubleDispatchVisitor.java (hunk 2 above)
Comment (vilchik-elena) [inline javascript-frontend/src/main/java/org/sonar/plugins/javascript/api/visitors/DoubleDispatchVisitor.java]: @pynicolas indeed. Fixed in 0319ebf51c4aef3d9f71d29ff92779d3ffeea59c

**Pull request merged**